var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -357,181 +326,179 @@ def setup_build_environment(self, env):
             env.append_flags("CXXFLAGS", self.spec["llvm-openmp"].headers.include_flags)
             env.append_flags("LDFLAGS", self.spec["llvm-openmp"].libs.ld_flags)
 
-    # Get any recent versions or non-numeric version
-    # Note that develop > numeric and non-develop < numeric
+    def _get_sys_type(self, spec):
+        sys_type = spec.architecture
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        return sys_type
 
-    @when("@:0.90,0.94:")
-    def cmake_args(self):
-        spec = self.spec
-        args = self.common_config_args
-        args.extend(
-            [
-                "-DCMAKE_CXX_STANDARD=17",
-                "-DLBANN_WITH_CNPY=%s" % ("+numpy" in spec),
-                "-DLBANN_DETERMINISTIC:BOOL=%s" % ("+deterministic" in spec),
-                "-DLBANN_WITH_HWLOC=%s" % ("+hwloc" in spec),
-                "-DLBANN_WITH_ALUMINUM:BOOL=%s" % ("+al" in spec),
-                "-DLBANN_WITH_ADDRESS_SANITIZER:BOOL=%s" % ("+asan" in spec),
-                "-DLBANN_WITH_BOOST:BOOL=%s" % ("+boost" in spec),
-                "-DLBANN_WITH_CONDUIT:BOOL=%s" % ("+conduit" in spec),
-                "-DLBANN_WITH_NVSHMEM:BOOL=%s" % ("+nvshmem" in spec),
-                "-DLBANN_WITH_FFT:BOOL=%s" % ("+fft" in spec),
-                "-DLBANN_WITH_ONEDNN:BOOL=%s" % ("+onednn" in spec),
-                "-DLBANN_WITH_ONNX:BOOL=%s" % ("+onnx" in spec),
-                "-DLBANN_WITH_EMBEDDED_PYTHON:BOOL=%s" % ("+python" in spec),
-                "-DLBANN_WITH_PYTHON_FRONTEND:BOOL=%s" % ("+pfe" in spec),
-                "-DLBANN_WITH_ROCTRACER:BOOL=%s" % ("+rocm +distconv" in spec),
-                "-DLBANN_WITH_TBINF=OFF",
-                "-DLBANN_WITH_UNIT_TESTING:BOOL=%s" % ("+unit_tests" in spec),
-                "-DLBANN_WITH_VISION:BOOL=%s" % ("+vision" in spec),
-                "-DLBANN_WITH_VTUNE:BOOL=%s" % ("+vtune" in spec),
-                "-DLBANN_DATATYPE={0}".format(spec.variants["dtype"].value),
-                "-DCEREAL_DIR={0}".format(spec["cereal"].prefix),
-                # protobuf is included by py-protobuf+cpp
-                "-DProtobuf_DIR={0}".format(spec["protobuf"].prefix),
-                "-Dprotobuf_MODULE_COMPATIBLE=ON",
-            ]
+    @property
+    def cache_name(self):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            hostname = hostname.rstrip("1234567890")
```

Comment:
I'm guessing this is done for builds conducted from allocations (i.e. where the nodes get a number): I recommend making a comment about that (in particular note that it is entirely for cosmetic reasons).

Reply:
All nodes on LC machines get numbers, even the login and/or launch nodes. This is just to get back to the canonical system name (e.g., `pascal` or `corona` or `lassen`).